### PR TITLE
Encoder Graph added

### PR DESCRIPTION
**Pull request by abhinavGirish**
to graph encoder data live

### CODE REVIEW

**Comment by abhinavGirish** [comment]
don't know why Travis is failing - would normally build if you have the correct libraries

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
+package com.roboclub.robobuggy.ui;
+import java.util.Random;
+
+import javax.swing.JPanel;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java]: Codacy Issue found: [Avoid unused imports such as 'javax.swing.JPanel'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833533/issues/source?bid=2938665&fileBranchId=3004589#l4)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
+import com.roboclub.robobuggy.messages.SteeringMeasurement;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java]: Codacy Issue found: [Avoid unused imports such as 'com.roboclub.robobuggy.messages.SteeringMeasurement'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833533/issues/source?bid=2938665&fileBranchId=3004589#l7)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java [hunk 3]
```diff
@@ -0,0 +1,49 @@
+package com.roboclub.robobuggy.ui;
+import java.util.Random;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java]: Codacy Issue found: [Avoid unused imports such as 'java.util.Random'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833533/issues/source?bid=2938665&fileBranchId=3004589#l2)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
+import com.roboclub.robobuggy.messages.SteeringMeasurement;
+import com.roboclub.robobuggy.ros.Message;
+import com.roboclub.robobuggy.ros.MessageListener;
+import com.roboclub.robobuggy.ros.SensorChannel;
+import com.roboclub.robobuggy.ros.Subscriber;
+
+import org.jfree.chart.*;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.data.xy.XYDataset;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/EncoderGraph.java]: Codacy Issue found: [Avoid unused imports such as 'org.jfree.data.xy.XYDataset'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833533/issues/source?bid=2938665&fileBranchId=3004589#l15)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/logging/SensorLogger.java [hunk 5]
```diff
@@ -34,7 +34,7 @@
 	private final Queue<String> _logQueue;
 	private final ArrayList<Subscriber> subscribers;
 
-	private static final Queue<String> startLoggingThread(PrintStream stream) {
+	private static final Queue<String> startLoggingThread(final PrintStream stream) {
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/logging/SensorLogger.java]: Codacy Issue found: [Avoid really long methods.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833474/issues/source?bid=2938665&fileBranchId=3004589#l37)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java [hunk 6]
```diff
@@ -42,6 +42,7 @@ public double getLongitude(){
 	private int frameWidth;
 	private int frameHeight;
 	private Subscriber gpsSub;
+	private Subscriber encSub;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused private fields such as 'encSub'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1502833534/issues/source?bid=2938665&fileBranchId=3004589#l45)

**Comment by TrevorDecker** [comment]
This looks really good
Can you flip the x and y axis? 
Also can you embed the graph inside the main gui instead of having it as a separate window?

**Comment by VivaanBahl** [comment]
I asked him to put it in a separate window so that none of the other components would get compressed. If we want it in the same window, I would vote to get rid of the GraphPanel (all the dials that don't do anything) and put it there

**Comment by TrevorDecker** [comment]
I am ok with removing replacing one of the Graph Panel graphs.  I would really like to be able to view data from all of the sensors at once.  At rolls I would rather not have to switch between windows to check data. 

I do like the large graph and we should keep the code for it, so that we can reenable it during debugging.

**Comment by abhinavGirish** [comment]
Okay I guess we'll work on embedding sometime later...gotta study for
finals now though :P

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java (hunk 6 above)
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Avoid unused private fields such as 'encSub'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559721999/issues/source?bid=2938665&fileBranchId=3004589#l45)

Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GpsPanel.java]: Codacy Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559721999/issues/source?bid=2938665&fileBranchId=3004589#l45)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 7]
```diff
@@ -4,11 +4,18 @@
...
 import com.roboclub.robobuggy.messages.SteeringMeasurement;
 import com.roboclub.robobuggy.ros.Message;
 import com.roboclub.robobuggy.ros.MessageListener;
 import com.roboclub.robobuggy.ros.SensorChannel;
 import com.roboclub.robobuggy.ros.Subscriber;
+import org.jfree.chart.*;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.data.xy.XYDataset;
+import org.jfree.data.xy.XYSeries;
+import org.jfree.data.xy.XYSeriesCollection;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused imports such as 'org.jfree.data.xy.XYSeriesCollection'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l18)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 8]
```diff
@@ -47,5 +54,16 @@ public void actionPerformed(String topicName, Message m) {
 				// TODO handle imu updates for graphs
 			}
 		});
+		
+        new Subscriber(SensorChannel.ENCODER.getMsgPath(), new MessageListener() {
+			
+			public void actionPerformed(String topicName, Message m) {
+				EncoderMeasurement em = (EncoderMeasurement)m;
+				double val1 = ((EncoderMeasurement)m).distance;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused local variables such as 'val1'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l62)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 9]
```diff
@@ -4,11 +4,18 @@
...
 import javax.swing.JPanel;
+
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
 import com.roboclub.robobuggy.messages.SteeringMeasurement;
 import com.roboclub.robobuggy.ros.Message;
 import com.roboclub.robobuggy.ros.MessageListener;
 import com.roboclub.robobuggy.ros.SensorChannel;
 import com.roboclub.robobuggy.ros.Subscriber;
+import org.jfree.chart.*;
+import org.jfree.chart.plot.PlotOrientation;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused imports such as 'org.jfree.chart.plot.PlotOrientation'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l15)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 10]
```diff
@@ -4,11 +4,18 @@
...
+import org.jfree.data.xy.XYDataset;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused imports such as 'org.jfree.data.xy.XYDataset'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l16)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 11]
```diff
@@ -47,5 +54,16 @@ public void actionPerformed(String topicName, Message m) {
 				// TODO handle imu updates for graphs
 			}
 		});
+		
+        new Subscriber(SensorChannel.ENCODER.getMsgPath(), new MessageListener() {
+			
+			public void actionPerformed(String topicName, Message m) {
+				EncoderMeasurement em = (EncoderMeasurement)m;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused local variables such as 'em'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l61)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 12]
```diff
@@ -4,11 +4,18 @@
...
+import com.roboclub.robobuggy.messages.EncoderMeasurement;
 import com.roboclub.robobuggy.messages.SteeringMeasurement;
 import com.roboclub.robobuggy.ros.Message;
 import com.roboclub.robobuggy.ros.MessageListener;
 import com.roboclub.robobuggy.ros.SensorChannel;
 import com.roboclub.robobuggy.ros.Subscriber;
+import org.jfree.chart.*;
+import org.jfree.chart.plot.PlotOrientation;
+import org.jfree.data.xy.XYDataset;
+import org.jfree.data.xy.XYSeries;
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused imports such as 'org.jfree.data.xy.XYSeries'](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l17)

File: real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java [hunk 13]
```diff
@@ -47,5 +54,16 @@ public void actionPerformed(String topicName, Message m) {
 				// TODO handle imu updates for graphs
 			}
 		});
+		
+        new Subscriber(SensorChannel.ENCODER.getMsgPath(), new MessageListener() {
+			
+			public void actionPerformed(String topicName, Message m) {
+				EncoderMeasurement em = (EncoderMeasurement)m;
+				double val1 = ((EncoderMeasurement)m).distance;
+				double val2 = ((EncoderMeasurement)m).timestamp.getTime();
```
Comment (TrevorDecker) [inline real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/GraphPanel.java]: Codacy Issue found: [Avoid unused local variables such as 'val2'.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1559722000/issues/source?bid=2938665&fileBranchId=3004589#l63)

**Comment by TrevorDecker** [comment]
What is the status of 
flipping the x and y axis? 
Also can you embed the graph inside the main gui instead of having it as a separate window?

**Pull request merged**